tests/fixtures/lxd-test.ts

```diff
@@ -1,23 +1,24 @@
-import { test as base } from "@playwright/test";
+import { Page, test as base } from "@playwright/test";
 import { finishCoverage, startCoverage } from "./coverage";
 
 export type LxdVersions = "5.0-stable" | "latest-edge";
 export type TestOptions = {
   lxdVersion: LxdVersions;
   hasCoverage: boolean;
+  runCoverage: Page;
 };
 
 export const test = base.extend<TestOptions>({
   lxdVersion: ["latest-edge", { option: true }],
   hasCoverage: [false, { option: true }],
-});
-
-test.beforeEach(async ({ page, hasCoverage }) => {
-  await startCoverage(page, hasCoverage);
-});
-
-test.afterEach(async ({ page, hasCoverage }) => {
-  await finishCoverage(page, hasCoverage);
+  runCoverage: [
```

Comment:
so I found the issue wasn't with nyc, turns out global `beforeEach` and `afterEach` hooks in playwright should be done this way instead of using `test.beforeEach` and `test.afterEach`. This was discovered after debugging where `finishCoverage` didn't run for every test.